src/test/java/com/sgordievskiy/spring/controller/TodoControllerIT.java

```diff
@@ -0,0 +1,247 @@
+    LocalDateTime date = LocalDateTime.of(2024, 11, 7, 12, 15, 30);
+    TodoUpdateDto requestDto = new TodoUpdateDto(
+      1L,
+      "New title",
+      "Description",
+      date,
+      Priority.MEDIUM,
+      Status.IN_PROGRESS
+    );
+
+    TodoResponseDto expected = new TodoResponseDto(
+      1L,
+      "New title",
+      "Description",
+      date,
+      Priority.MEDIUM,
+      Status.IN_PROGRESS,
+      date,
+      date,
+      1L
+    );
+
+    String jsonRequest = objectMapper.writeValueAsString(requestDto);
+
+    MvcResult result = mockMvc.perform(
+        put("/todos/1")
+          .content(jsonRequest)
+          .contentType(MediaType.APPLICATION_JSON)
+      )
+      .andExpect(status().isOk())
+      .andReturn();
+
+    TodoResponseDto actual = objectMapper.readValue(result.getResponse().getContentAsString(), TodoResponseDto.class);
+    Assertions.assertNotNull(actual);
+    Assertions.assertNotNull(actual.getId());
+    Assertions.assertEquals(expected.getId(), actual.getId());
+    Assertions.assertEquals(expected.getTitle(), actual.getTitle());
+    Assertions.assertEquals(expected.getDescription(), actual.getDescription());
+    Assertions.assertEquals(expected.getDueDate(), actual.getDueDate());
+    Assertions.assertEquals(expected.getPriority(), actual.getPriority());
+    Assertions.assertEquals(expected.getStatus(), actual.getStatus());
+  }
+
+  @Test
+  @DisplayName("Delete existing task")
+  @Sql(
+    scripts = "classpath:database/restore-deleted-task.sql",
+    executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD
+  )
+  void deleteTodo_Success() throws Exception {
+    MvcResult result = mockMvc.perform(
+        delete("/todos/1")
+          .contentType(MediaType.APPLICATION_JSON)
+      )
+      .andExpect(status().isOk())
+      .andReturn();
+  }
+
+  @Test
+  @DisplayName("Get all tasks")
+  void getAll_ShouldReturnAllTasks() throws Exception {
+    LocalDateTime date = LocalDateTime.of(2024, 11, 5, 1, 2, 3);
+    List<TodoResponseDto> expected = new ArrayList<>();
+    expected.add(new TodoResponseDto(
+      1L,
+      "Task",
+      "",
+      date,
+      Priority.HIGH,
+      Status.PENDING,
+      date,
+      date,
+      1L
+    ));
+
+    // When
+    MvcResult result = mockMvc.perform(
+        get("/todos")
+          .contentType(MediaType.APPLICATION_JSON)
+      )
+      .andExpect(status().isOk())
+      .andReturn();
+
+    // Then
+    TodoResponseDto[] actual = objectMapper.readValue(result.getResponse().getContentAsByteArray(), TodoResponseDto[].class);
+    Assertions.assertEquals(1, actual.length);
+    Assertions.assertEquals(expected, Arrays.stream(actual).toList());
+  }
+
+  @Test
+  @DisplayName("Get task history")
+  void getTaskHistory_ShouldReturnTaskHistory() throws Exception {
+    LocalDateTime date = LocalDateTime.of(2024, 11, 5, 1, 2, 3);
+    List<TaskHistoryResponseDto> expected = new ArrayList<>();
+    expected.add(new TaskHistoryResponseDto(
```

Comment:
Optional: you can use List.of():
```suggestion
    var taskHistoryResponseDto = new TaskHistoryResponseDto(
    
    ....
    );
    List<TaskHistoryResponseDto> expected = List.of(taskHistoryResponseDto);
```